### PR TITLE
TagGroup: add docs for help text, contextual help, and avatars

### PR DESCRIPTION
**Pull request by reidbarber**
## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

RSP

### CODE REVIEW

**Review by ktabors — DISMISSED** [review]

**Review by LFDanLu — DISMISSED** [review]
LGTM

**Comment by devongovett** [comment]
Is it right that the label doesn't align with the first tag?

image

**Comment by devongovett** [comment]
Also the label should not use a `<label>` element, because it isn't labeling an HTML form element. Should just be a `<span>`.

File: packages/@react-spectrum/tag/docs/TagGroup.mdx [hunk 1]
```diff
@@ -154,3 +164,113 @@ function Example() {
   </Item>
 </TagGroup>
 ```
+
+### With avatars
+
+```tsx example
+<TagGroup aria-label="TagGroup with avatars example">
+  <Item>
+    <Avatar src="https://i.imgur.com/kJOwAdv.png" alt="Person 1 avatar" />
```
Comment (devongovett) [inline packages/@react-spectrum/tag/docs/TagGroup.mdx:173]: I think these avatars should not have alt text (it defaults to "" in the Avatar component). Right now tags are read by screen readers as "Person 1 avatar Person 1", which is kinda unnecessarily verbose. The avatar is essentially decorative so I don't think it needs its own alt text.

**Comment by reidbarber** [comment]
> Is it right that the label doesn't align with the first tag?

I agree, it does look weird. I can fix by changing the margin from 4px on each side, to 8px on end.

**Comment by devongovett** [comment]
Or maybe the tags shouldn't have margins and we could use gap instead? Then the first/last tag wouldn't have a margin.

**Comment by reidbarber** [comment]
@devongovett We're using `display: inline` on the tags so that the actions can be on the same line, while being outside of the tag group. Gap only applies to flex, grid, and columns. Looking into a way to do both.

**Comment by devongovett** [comment]
Can you use `inline-flex` with `flex-wrap`?

**Comment by reidbarber** [comment]
Looks like that might work if I also make the actions inline-flex. Double checking for any side effects.

**Comment by reidbarber** [comment]
Pushed a fix in https://github.com/adobe/react-spectrum/pull/4102

**Comment by devongovett** [comment]
FYI looks like the example in the "Content" section overflows horizontally on mobile causing page scrolling.

**Comment by reidbarber** [comment]
@devongovett I can remove the Flex's used in these; don't think they're adding much.

Should I replace the Adobe products tags with News, Shopping, etc? May be closer to real life use case and have more realistic width for wrapping on mobile.

**Comment by devongovett** [comment]
Sure 🤷

**Review by devongovett — APPROVED** [review]

File: packages/@react-spectrum/tag/docs/TagGroup.mdx [hunk 2]
```diff
@@ -154,3 +147,104 @@ function Example() {
...
+    {id: 4, name: 'Shopping'}
+  ];
+
+  let [items, setItems] = React.useState(defaultItems);
+
+  let removeItem = (key) => {
+    setItems(prevItems => prevItems.filter((item) => key !== item.id));
+  };
+
+  let isValid = items.length <= 8;
```
Comment (LFDanLu) [inline packages/@react-spectrum/tag/docs/TagGroup.mdx:211]:
```suggestion
  let isValid = items.length <= 3;
```

**Review by LFDanLu — CHANGES_REQUESTED** [review]

**Review by LFDanLu — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Review by devongovett — APPROVED** [review]

**Pull request merged**